enterprise/server/usage/usage.go

```diff
@@ -100,20 +106,16 @@ func NewTracker(env environment.Env, opts *TrackerOpts) (*tracker, error) {
 	if clock == nil {
 		clock = timeutil.NewClock()
 	}
-
-	pool := redsync_goredis.NewPool(rdb)
-	rs := redsync.New(pool)
-	flushMutex := rs.NewMutex(
-		redisUsageLockKey,
-		redsync.WithTries(1), redsync.WithExpiry(redisUsageLockExpiry),
-	)
-
+	flushLock := opts.FlushLock
+	if flushLock == nil {
```

Comment:
nit: this conditional logic is still error-prone. If you accept the lock interface as a param and your impl just always uses it, then we'll be good